res/controllers/M-Audio-Xponent-Advanced-scripts.js

```diff
@@ -296,7 +296,33 @@ MaudioXponent.syncLights = function() {
         engine.trigger(deck.group, "loop_out");
 
         midi.sendShortMsg(deck.on, MaudioXponent.buttons.scratch, deck.scratchEnabled);
-        MaudioXponent.onPlayPositionChange(deck.playPosition, deck.group);
+        
+        var duration = engine.getValue(deck.group, "duration");
+        deck.warnAt = (duration - 30) / parseFloat(duration);
+        engine.trigger(deck.group, "playposition");
+    }
+};
+
+MaudioXponent.syncEffects = function() {
+    var deck = MaudioXponent.rightDeck;
+
+    for (var i = 1; i <= 4; i++) {
+        var control = MaudioXponent.buttons.fx1 + i - 1;
+        var group = "[EffectRack1_EffectUnit" + i + "]";
+        if (!deck.shift) {
+            // Show enabled effect units
+            var enabled = engine.getValue(group, "enabled");
+            midi.sendShortMsg(deck.on, control, enabled);
+            engine.setValue(group, "show_focus", false);
+        } else {
+            // Show focused effect unit
+            var focused = MaudioXponent.state.focusedEffectUnit === i;
+            midi.sendShortMsg(deck.on, control, focused);
```

Comment:
I could possibly have the lights remain as they are up until you actually push one of them to focus on an effect unit, which would prevent them from switching over every time you press right shift, perhaps to do something other than effects, but that seems confusing since it might imply that multiple effects are focused. The Xponent's lights are not terribly bright, so it's not as jarring as you might think. I'll give it a try and see what it feels like.